src/util/memory.rs

```diff
@@ -192,17 +193,13 @@ pub fn get_process_memory_maps() -> String {
 }
 
 /// Returns the total physical memory for the system in bytes.
-pub(crate) fn get_system_total_memory() -> usize {
-    match sys_info::mem_info() {
-        Ok(mem_info) => mem_info.total as usize,
-        Err(e) => {
-            warn!(
-                "Failed to get sys_info::mem_info: {:?}. Return 1G in get_system_total_memory()",
-                e
-            );
-            1024 * 1024 * 1024
-        }
-    }
+pub(crate) fn get_system_total_memory() -> u64 {
```

Comment:
Changing to `u64` is fine. However, this suggests we should not use `get_system_total_memory()` for setting a default heap size. Maybe we should have something like [`get_contraimed_memory()`](https://docs.rs/libuv/latest/libuv/misc/fn.get_constrained_memory.html) instead (comparing to [`get_total_memory()`](https://docs.rs/libuv/latest/libuv/misc/fn.get_total_memory.html)).

Reply:
Ah right. The main issue is that you can run 32-bit binaries on 64-bit computers. So you generally have more than 4 GB memory visible to a 32-bit build. Maybe we can just have a compare against 4 GB and if more then cap it to 4 GB for 32-bit builds?

Reply:
Actually you're right. `get_constrained_memory()` is a better solution as one can reduce the memory limit for a specific process and MMTk should respect that.

Reply:
I think the current PR is fine. It changes the use of `sys_info_rs` to `sysinfo` which works with more platforms. However, if you are interested to do more changes in this PR, that sounds good.

Reply:
I might make a separate PR for that later if I work on it. I just wanted to quickly remove the sys-info-rs dependency as its build step is a bit non-standard.